server/src/tests/integration/erc20.rs

```diff
@@ -239,9 +239,14 @@ pub async fn l2_erc20_balance_of(token: Address, account: Address, rpc_url: &str
         .call()
         .await?;
 
+    // Extract the first result and combine with EVM result layout for deserialization
+    let output: Vec<Vec<u8>> = bcs::from_bytes(&eth_call_result)?;
     let return_values = SerializedReturnValues {
         mutable_reference_outputs: Vec::new(),
-        return_values: bcs::from_bytes(&eth_call_result)?,
+        return_values: Vec::from([(
+            output.first().unwrap().to_owned(),
+            EVM_NATIVE_OUTCOME_LAYOUT.clone(),
+        )]),
```

Comment:
My only concern is whether we would have to keep supplying the layout still, despite not having it on the client side for some internal things like this. And whether such spots would grow when we add EVM NFT support etc.

Reply:
I checked this and we only do this in the integration test. Other tests perform the `session.execute_function_bypass_visibility()` calls directly without going through the `call` API. So I think this won't be an issue in general.